test/service/test_manifest.py

```diff
@@ -1089,41 +1099,50 @@ def test_manifest_content_disposition_header(self, _can_use_cached_manifest):
 
     @mock_sts
     @mock_s3
-    def test_full_hash_validity(self):
+    def test_hash_validity(self):
         self.maxDiff = None
         old_bundle = ("aaa96233-bf27-44c7-82df-b4dc15ad4d9d", "2018-11-02T113344.698028Z")
         self._index_canned_bundle(old_bundle)
-
-        # When a new bundle is indexed and its full manifest cached,
-        # a matching object_key is generated ...
         filters = {'project': {'is': ['Single of human pancreas']}}
-        format_ = 'full'
-
+        old_object_keys = {}
         service = ManifestService(StorageService())
-        generator = manifest_service.FullManifestGenerator(service, filters)
-        old_bundle_object_key = service._derive_manifest_key(format_, filters, generator.manifest_content_hash)
-
-        # and should remain valid ...
-        new_generator = manifest_service.FullManifestGenerator(service, filters)
-        self.assertEqual(old_bundle_object_key,
-                         service._derive_manifest_key(format_, filters, new_generator.manifest_content_hash))
-
-        # ... until a new bundle belonging to the same project is indexed, at which point a manifest request
-        # will generate a different object_key ...
-        new_bundle = ("aaa96233-bf27-44c7-82df-b4dc15ad4d9d", "2018-11-04T113344.698028Z")
-        self._index_canned_bundle(new_bundle)
-        generator = manifest_service.FullManifestGenerator(service, filters)
-        new_bundle_object_key = service._derive_manifest_key(format_, filters, generator.manifest_content_hash)
-
-        # ... invalidating the cached object previously used for the same filter.
-        self.assertNotEqual(old_bundle_object_key, new_bundle_object_key)
-
-        # Updates or additions, unrelated to that project do not affect object key generation
-        self._index_canned_bundle(("f79257a7-dfc6-46d6-ae00-ba4b25313c10", "2018-09-14T133314.453337Z"))
-        generator = manifest_service.FullManifestGenerator(service, filters)
-        latest_bundle_object_key = service._derive_manifest_key(format_, filters, generator.manifest_content_hash)
-
-        self.assertEqual(latest_bundle_object_key, new_bundle_object_key)
+        for format_ in ['compact', 'full', 'terra.bdbag']:
```

Comment:
Why is this done in three smaller loops rather than just one big loop?

Reply:
It has to do with indexing. In the iteration for the first format, the indexer would go through all object key update scenarios (old bundle, new bundle, non related bundle). Leaving the consecutive formats with a non updatable index. Unless the index is cleared at the end of each iteration.